modules/ctl/src/main/scala/hi/Common.scala

```diff
@@ -0,0 +1,45 @@
+package gem.ctl.hi
+
+import gem.ctl.free.ctl._
+import gem.ctl.low.git._
+import gem.ctl.low.docker._
+
+import scalaz._, Scalaz._
+
+/** Constructors for some common `CtlIO` operations that are shared by other commands. */
+object common {
+
+  def getUniqueRunningContainerWithLabel(label: String): CtlIO[Container] =
+    findRunningContainersWithLabel(label) flatMap {
+      case c :: Nil => c.point[CtlIO]
+      case Nil      => error("No running container found.")       *> exit(-1)
+      case _        => error("Multiple running container found.") *> exit(-1)
+    }
+
+  def getRunningGemContainer: CtlIO[Container] =
+    gosub("Finding current running Gem container.") {
+      getUniqueRunningContainerWithLabel("edu.gemini.gem")
+    }
+
+  def getRunningPostgresContainer: CtlIO[Container] =
+    gosub("Finding current running Postgres container.") {
+      getUniqueRunningContainerWithLabel("edu.gemini.db")
+    }
+
+  def getDeployCommitForContainer(k: Container): CtlIO[DeployCommit] =
```

Comment:
It's a commit hash along with a tag that indicates whether there are uncommitted files or not. So if you deploy a version with local changes you can distinguish it from the HEAD commit.

```scala
case class DeployCommit(commit: Commit, uncommitted: Boolean) {
  def imageVersion = if (uncommitted) s"${commit.hash}-UNCOMMITTED" else commit.hash
}
```